metric/prometheus.go

```diff
@@ -107,19 +143,40 @@ func GetName(fields ...string) string {
 	return name
 }
 
-// ValidName check if the char is valid
-func ValidName(ch byte) bool {
-	return (ch >= 'a' && ch <= 'z') || (ch >= 'A' && ch <= 'Z') || (ch >= '0' && ch <= '9') || ch == '_' || ch == '-'
+// ValidMetricName check if the metric name is valid
+func ValidMetricName(name string) bool {
+	return validMetric.MatchString(name)
+}
+
+// ValidLabelName check if the label name is valid
+func ValidLabelName(label string) bool {
+	return validLabel.MatchString(label)
+}
+
+// ValidMetricChar check if the char is valid for metric name
+func ValidMetricChar(ch byte) bool {
+	return (ch >= 'a' && ch <= 'z') || (ch >= 'A' && ch <= 'Z') ||
+		(ch >= '0' && ch <= '9') || ch == '_' || ch == '-' || ch == ':'
 }
 
 // RemoveInvalidChars remove invalid chars
 func RemoveInvalidChars(name string) string {
 	var result []byte
-	for i := 0; i < len(name); i++ {
-		if ValidName(name[i]) {
-			result = append(result, name[i])
+	i := 0
+
+	// skip all of the non-alphabetic chars
```

Comment:
what is the purpose of the `skip for` code segment ?

Reply:
because the leading number is invalid!